tests/test_checkout.py

```diff
@@ -0,0 +1,41 @@
+import shutil
```

Comment:
Let's keep benchmarks in a separate directory (e.g. tests/benchmarks or smth).

Reply:
I think it's convenient to have benchmarks + tests related to an API in one single place. We can decide when we have more benchmarks. :)

Reply:
I mean the stage we are in right now with the tests, it's easier to share small fixtures within a module/testfile than globally.

Reply:
It will be really painful to separate them. Let's keep them separate right away, so we don't have the pain later. It doesn't cost us anything to properly separate them right now.

Reply:
Let's do this later when it starts to become painful. We need to evolve the test fixtures and benchmarks fixtures.

> It doesn't cost us anything to properly separate them right now.

It's not convenient compared to having in one place. Think of microbenchmarks, eg: loading trees/building trees, etc. It's convenient to have it in one place for all tests regarding loading objects than spread across different files.
Let's not over do here from the start.

Reply:
@skshetry If it becomes painful - it is too late :D We can't clean up our dvc test base right now and I don't want to have this mess again in this new project. Doing it correctly doesn't cost us anything - so let's do it now, it is obvious that the tests and benchmarks will grow and we are just waiting to have problems. This is not really "overdoing", as it costs us nothing to start with correct approach.

Microbenchmarks should be separate too. From the projects I've seen so far, benchmarks are usually separate and I totally understand why (esp after writing dvc-bench benchmarks). E.g. https://github.com/samuelcolvin/pydantic-core/tree/main/tests

Reply:
It should be easy to fix benchmarks than tests, as there should be few of them. I'd very much prefer handling them when it starts becoming painful or as our requirements change than starting with a complicated process/structure when I am adding a single test.

> We can't clean up our dvc test base right now

Benchmarks and tests like other parts of our code should be evolved together. We never fixed our tests for more than 3 years, whereas the code went through many major refactors. It's our own negligence to be honest, that we should not get away with the file hierarchy.

From my experience with `dvc-bench`, I don't agree that a deeply-nested approach is better than flat structure. Again. I'd prefer the requirements drive this and evolve accordingly.

Reply:
@skshetry I'm not proposing a deeply nested structure like in `dvc-bench`, I'm only asking to put benchmarks separately into `tests/benchmarks`, so that we have it properly structured from the very beginning. Because we both know that that it is just a matter of time until we have a mess on our hands with tests and benchmarks mixed together.  We spend more time arguing about this, than it would take to correctly structure the tests and just move on with this.

Reply:
It's still one level though, and that isolates benchmarks. The kind of benchmarks that I have in mind is simple and very much close to the tests that we write, unlike the ones in dvc-bench, and I don't want benchmarks to be treated differently than our tests with different fixtures, etc.

I'd prefer a much flatter approach, if we don't want to keep them in same file, we can keep them as `bench_*` files.
I understand your concern, I really do, there are many good reasons to structure as `tests/benchmarks`, eg: having isolated benchmark fixtures, real s3 tests, more e2e tests, etc. But I am not convinced that we should start with that and that are the things that I am trying to avoid for now as direction is unclear about `dvc-data` benchmarks, eg: if it'll be doing end-to-end benchmarks replacing part of dvc-bench, etc.

As for now, I'd prefer we keep things simple. We can reconsider this when we have 10 or more benchmarks, or when some benchmark really need us to rethink about the hierarchy. :)

Reply:
What's the difference between `bench_` and `benchmark/`? Same way of separating, but the latter is a conventional one that we have in dvc-bench already and is widely used in repos that use pytest-benchmark. It allows for easy scanning, so if anyone will look into a repo he will see that there are benchmarks and what functionality is rougly covered. Keeping it simple and conventional is putting it into `benchmarks` right away and not have the problem of having to go through needless waste of time separating them when it will become a problem.

Reply:
`bench_` is just intermediate to `benchmark/`. I am not convinced that we need to start with `benchmark/`, and just proposed an intermediate solution for the future before going to `benchmark/`. Also, I think you are overstating that it'll waste time. :)